src/v/storage/segment.h

```diff
@@ -295,6 +298,14 @@ inline size_t segment::size_bytes() const {
 inline bool segment::has_compaction_index() const {
     return _compaction_index != std::nullopt;
 }
+
+inline bool
+segment::has_compactible_offsets(const compaction_config& cfg) const {
+    // since we don't support partially-compacted segments, a segment must
+    // end before the max compactible offset to be eligible for compaction.
+    return offsets().stable_offset <= cfg.max_collectible_offset;
```

Comment:
would it possible that we want dirty offset since it's generally going to be the highest possible offset if writes happen to be in flight?

Reply:
I thought we only compact closed segments?